src/load_order.ts

```diff
@@ -587,16 +582,28 @@ export const startREDmodDeployInTheBackgroundWithNotifications = (
 
   // Should really figure out how to get the output from this.
   // TODO: https://github.com/E1337Kat/cyberpunk2077_ext_redux/issues/321
+  const modListPath = path.join(gameDirPath, V2077_MODLIST_PATH);
   const REDdeployment: Promise<void> =
-    vortexApi.runExecutable(redDeploy.executable, redDeploy.args, redDeploy.options)
-      .then(() => {
-        vortexApi.log(`info`, `${me}: REDmod deployment ${loID} complete!`);
-        showInfoNotification(vortexApi, InfoNotification.REDmodDeploymentSucceeded);
-      })
-      .catch((error) => {
-        vortexApi.log(`error`, `${me}: REDmod deployment ${loID} failed!`, S(error));
-        showInfoNotification(vortexApi, InfoNotification.REDmodDeploymentFailed);
-      });
+    pipe(
+      loadOrderToREDdeployModList(v2077LoadOrderToDeploy),
+      (generatedModList) => generatedModList.join(`\n`),
+      (encodedLoadOrder) => tryCatchTE(
+        () =>
+          fs.statAsync(path.dirname(modListPath)).then(() =>
+            fs.writeFileAsync(`${modListPath}.${loID}.tmp`, encodedLoadOrder, { encoding: `utf8` })).then(() =>
+            fs.renameAsync(`${modListPath}.${loID}.tmp`, modListPath)).then(() => true),
+        (error) => new Error(`Unable to write load order to disk: ${S(error)}`),
+      ),
+    )().then(() =>
+      vortexApi.runExecutable(redDeploy.executable, redDeploy.args, redDeploy.options)
```

Comment:
This is slightly invisible here, but promise chains are always the same.. we can refactor this into a `TaskEither` in the next pass.